File: libs/eave-stdlib-ts/src/core-api/integrations/github.ts
import { sharedConfig } from '../../config.js';
import { RequestArgsOrigin, makeRequest } from '../../lib/requests.js';
import { Team } from '../team.js';

export type GithubInstallationInput = {
  github_install_id: string;
}

export type GithubInstallation = {
  id: string;
  /** eave TeamOrm model id */
  team_id: string;
  github_install_id: string;
}

export type GetGithubInstallationRequestBody = {
  github_integration: GithubInstallationInput;
}

export type GetGithubInstallationResponseBody = {
  team: Team;
  github_integration: GithubInstallation;
}

export async function getGithubInstallation({ origin, input }: RequestArgsOrigin & {input: GetGithubInstallationRequestBody}): Promise<GetGithubInstallationResponseBody> {
  const resp = await makeRequest({
    url: `${sharedConfig.eaveApiBase}/integrations/github/query`,
    origin,
    input,
  });
  const responseData = <GetGithubInstallationResponseBody>(await resp.json());
  return responseData;
}

File: libs/eave-stdlib-py/src/eave/stdlib/core_api/models/github.py
import uuid
from eave.stdlib.core_api.models import BaseInputModel, BaseResponseModel


import pydantic


class GithubInstallation(BaseResponseModel):
    id: uuid.UUID
    team_id: uuid.UUID
    """eave TeamOrm model id"""
    github_install_id: str

GithubInstallationPeek = GithubInstallation
"""Type Alias for naming consistency with other integrations."""

class GithubInstallationInput(BaseInputModel):
    github_install_id: str
